internal/core/metadata/rest_device.go

```diff
@@ -63,20 +63,13 @@ func restAddNewDevice(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Addressable check
-	// Try by name
-	//TODO: Is what we really need here a checkForAddressable() function?
-	addressable, err := dbClient.GetAddressableByName(d.Addressable.Name)
-	if err != nil {
-		// Try by ID
-		addressable, err = dbClient.GetAddressableById(d.Addressable.Id)
-		if err != nil {
-			LoggingClient.Error(err.Error())
-			http.Error(w, err.Error()+": A device must be associated to an Addressable", http.StatusBadRequest)
-			return
-		}
+	// Protocol check
+	if len(d.Protocols) == 0 {
```

Comment:
In case of device-random or device-simple, Protocols length can be 0. So maybe should be better to remove this check.

Reply:
I'll take this under advisement. In the case of device-random, it currently creates an addressable from the configuration.toml
```
[[DeviceList]]
  Name = "Random-Integer-Generator01"
  Profile = "Random-Integer-Generator"
  Description = "Example of Device Random"
  Labels = [ "device-random-example" ]
  [DeviceList.Addressable]
    Address = "device-random-01"
    Protocol = "OTHER"
```
Presumably the Device Service team is going to create a corollary for protocol entries from the configuration, and I expect they'll convert this one to populate some kind of dummy protocol entry.

@tonyespy @cloudxxx8 for visibility

Reply:
Currently device-random creates an addressable because otherwise it's not possible to create the device. As you said, with this check you are forcing to use a dummy protocol entry in some cases, which can be avoided.